### ISSUE TITLE
Add link-checking rule powered by `lychee-lib` (broken hyperlinks + mail addresses in Markdown)

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#279]
## Summary

`rumdl` would be much more useful for documentation CI if it could also detect broken hyperlinks and invalid/broken `mailto:` addresses inside Markdown files.

## Proposal

Add an opt-in rule (or rule group) that:

* Extracts links from Markdown
* Validates HTTP/HTTPS links and `mailto:` addresses
* Reports failures as standard `rumdl` diagnostics (file, line/column, message)

## Suggested implementation

Use `lychee-lib` as the engine:

* Prefer library integration (not shelling out to `lychee`) so it works consistently across CLI and LSP
* Build a configurable client via `ClientBuilder` (timeouts, retries, concurrency, headers, ignores)

Relevant upstream docs:

* https://github.com/lycheeverse/lychee?#library-usage
* https://lychee.cli.rs/guides/library
* https://docs.rs/lychee-lib/latest/lychee_lib

## Configuration ideas

* On by default, but using the `--offline`[Only check local files and block network requests] mode.
* Optional online mode (network checks can be slow / flaky) can be set in the rumdl.toml file.
* Settings to control:
  * Timeout and concurrency
  * Ignore list (domains/patterns)
  * Whether to check `mailto:` addresses
  * Cache support to avoid re-checking unchanged links

## Acceptance criteria

* Finds broken HTTP/HTTPS links in Markdown
* Validates `mailto:` addresses (and reports invalid/broken cases)
* Clear, actionable output, including the failing URL/address and reason
* Works with `rumdl check` and `rumdl check --fix` (no formatting changes needed, just diagnostics)

**Comment by rvben** [comment]
Thanks for the detailed proposal. I investigated integrating `lychee-lib` directly and here's what I found:

**Dependency impact:**
- lychee-lib adds ~125 new dependencies (756 total vs rumdl's current 695)
- Includes `octocrab` (full GitHub API client) as a **non-optional** dependency
- Estimated binary size increase: +3-5MB
- Significant compile time increase

The `octocrab` dependency is the main blocker—it's a heavy GitHub API client that can't be disabled, yet most users won't need GitHub API authentication for link checking.

**Recommendation: Run lychee alongside rumdl**

Both tools are fast Rust CLIs that complement each other well:

```yaml
# GitHub Actions example
- name: Lint markdown
  run: rumdl check docs/

- name: Check links
  run: lychee docs/**/*.md
```

This approach:
- Keeps rumdl lean and focused on markdown linting/formatting
- Lets you use lychee's full feature set (caching, ignore patterns, GitHub token support, etc.)
- Avoids bloating the binary for users who don't need link checking

Closing as out-of-scope for rumdl's core.
